evaluate/task/task-code-repair.go

```diff
@@ -0,0 +1,155 @@
+package task
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/symflower/eval-dev-quality/evaluate/metrics"
+	"github.com/symflower/eval-dev-quality/language"
+	"github.com/symflower/eval-dev-quality/log"
+	"github.com/symflower/eval-dev-quality/model"
+	evaltask "github.com/symflower/eval-dev-quality/task"
+)
+
+// TaskCodeRepair holds the code repair task.
+type TaskCodeRepair struct {
+	// ResultPath holds the directory path where results should be written to.
+	ResultPath string
+
+	// Language holds the language for which the task should be evaluated.
+	Language language.Language
+	// Model holds the model which the task should be evaluated.
+	Model model.Model
+
+	// Logger holds the logger for this tasks.
+	Logger *log.Logger
+}
+
+// TaskArgumentsCodeRepair holds extra arguments to be used in a query prompt.
+type TaskArgumentsCodeRepair struct {
+	// Mistakes holds the list of compilation errors for a repository.
+	Mistakes []string
+}
+
+var _ evaltask.Task = (*TaskCodeRepair)(nil)
+
+// newCodeRepairTask returns a code repair task.
+func newCodeRepairTask(logger *log.Logger, resultPath string, model model.Model, language language.Language) (task evaltask.Task) {
+	return &TaskCodeRepair{
+		ResultPath: resultPath,
+		Language:   language,
+		Model:      model,
+		Logger:     logger,
+	}
+}
+
+// Identifier returns the code repair task identifier.
+func (t *TaskCodeRepair) Identifier() evaltask.Identifier {
+	return IdentifierCodeRepair
+}
+
+// Run performs source code repairing in a repository with compilation errors.
```

Comment:
Please add a docstring saying how the "task cases" are extracted/iterated from a repository.